### DISCUSSION TITLE
API Usage with Minio?

### DISCUSSION THREAD

**Discussion by teaglin** [awslabs/aws-sdk-swift#1023] [Q&A] [answered]
I was interested in using the Swift s3 api to connect to Minio, but glancing through the api I couldn't find anything similar to the Python api. Is there anything like this in the current Swift implementation?

```
s3 = boto3.resource('s3',
                    endpoint_url='YOUR_URL',
                    aws_access_key_id='YOUR_ACCESS_KEY',
                    aws_secret_access_key='YOUR_SECRET_KEY',
                    config=Config(signature_version='s3v4'),
                    region_name='us-east-1')

```

**Comment by jbelkins** [comment, marked as answer]
I am not sure if this will work for you, but try creating your client this way & let me know if it works:
```
            let config = try S3Client.S3ClientConfiguration(region: "us-east-1", endpoint: "https://yours3endpoint.com/")
            let client = S3Client(config: config)
```
When performing operations on a bucket `yourbucket`, the connection will be made to `yourbucket.yours3endpoint.com`.

Officially supporting custom endpoints is on the feature list for this SDK but I do not have a timeframe for when it will be officially available.

**Reply by teaglin** [reply]
Thanks for the help! I think that should work. I got one more question. I am attempting to connect to a Minio deployment that doesn't support Https. Is there a way to disable TLS?

I tried setting the `httpClientConfiguration`
```
config.httpClientConfiguration = ClientRuntime.HttpClientConfiguration(protocolType: .http)
```

That doesn't seem to work. Digging through the `SdkHttpClient` it appears the config is not used.

```
public class SdkHttpClient {

    let engine: HttpClientEngine

    public init(engine: HttpClientEngine, config: HttpClientConfiguration) {
        self.engine = engine
    }

```

**Reply by jbelkins** [reply]
We do not support http connections at this time, although we do plan to in the future.  Unfortunately, I don't have a timeframe for when that feature will be available.
